src/algebra/big_operators.lean

```diff
@@ -37,6 +37,26 @@ protected def prod [comm_monoid β] (s : finset α) (f : α → β) : β := (s.1
 @[to_additive]
 theorem prod_eq_fold [comm_monoid β] (s : finset α) (f : α → β) : s.prod f = s.fold (*) 1 f := rfl
 
+end finset
+
+@[to_additive]
+lemma monoid_hom.map_prod [comm_monoid β] [comm_monoid γ] (g : β →* γ) (f : α → β) (s : finset α) :
+  g (s.prod f) = s.prod (λx, g (f x)) :=
+by simp only [finset.prod_eq_multiset_prod, g.map_multiset_prod, multiset.map_map]
+
+lemma ring_hom.map_prod [comm_semiring β] [comm_semiring γ]
+  (g : β →+* γ) (f : α → β) (s : finset α) :
+  g (s.prod f) = s.prod (λx, g (f x)) :=
+g.to_monoid_hom.map_prod f s
+
+lemma ring_hom.map_sum [comm_semiring β] [comm_semiring γ]
```

Comment:
Can't `to_additive` take care of this, after marking `g.to_add_monoid_hom` as the additive version of `g.to_monoid_hom`?

Reply:
I'd prefer to use `to_additive` only for straightforward transfer of multiplicative theory to additive theory, without rings involved. If you want to use it for rings, feel free to try, but it'll probably require attrs on `ring.to_monoid` etc as well.

Reply:
Ok, fair enough.